NDTensors/src/linearalgebra/linearalgebra.jl

```diff
@@ -424,7 +424,6 @@ matrix is unique. Returns a tuple (Q,R).
 function qr_positive(M::AbstractMatrix)
   sparseQ, R = qr(M)
   Q = convert(typeof(R), sparseQ)
-  Q = convert(typeof(R), sparseQ)
   nc = size(Q, 2)
   ## TODO issue here for GPU because tying to access indices
```

Comment:
Maybe we could add a function `iscu(A::AbstractArray) = leaf_parenttype(A) isa CuArray` and then throw an error at the beginning of these functions based on `iscu`, so users get a nice error message and it reminds us to fix it.